tests/test_bundle.py

```diff
@@ -332,6 +338,61 @@ async def test_disable_ingress_auth(ops_test: OpsTest):
     )
 
 
+async def test_gateway_replicas_config(ops_test: OpsTest):
+    """Test changing the replicas config to 2, and then:
+    1. Assert the replicas in the Deployment spec went up to 2.
+    2. Assert the Deployment has the antiaffinity rule as expected.
+    3. Assert the new Pod was not scheduled due to only 1 Node being available.
+    """
+
+    replicas_value = "2"
+    await ops_test.model.applications[ISTIO_GATEWAY_APP_NAME].set_config(
+        {"replicas": replicas_value}
+    )
+    await ops_test.model.wait_for_idle(apps=[ISTIO_GATEWAY_APP_NAME], status="active", timeout=300)
+
+    client = lightkube.Client()
+
+    # Get the gateway Deployment
+    gateway_deployment = client.get(
+        Deployment, name="istio-ingressgateway-workload", namespace=ops_test.model_name
+    )
+
+    # Assert the Deployment has the number of replicas from the config
+    assert gateway_deployment.spec.replicas == int(replicas_value)
+    # Assert the Deployment has the correct antiaffinity rule
```

Comment:
I wonder if it is actually worth checking this. I guess with this we are effectively testing that k8s is actually working.

Something we could do to ensure the manifests are correctly rendered in a less expensive test case is to unit test the gateway charm and inspect the rendered manifests this charm will apply [here](https://github.com/canonical/istio-operators/blob/main/charms/istio-gateway/src/charm.py#L99).

Reply:
I think here it is testing the config is being propagated and rendered correctly in the manifests, rather than testing k8s functionality. However, I do agree that we can convert this part to a unit test (there is no value in testing the rendering - this is basic jinja), and the rest of this integration test ensures the second Pod was not scheduled, so it would be good coverage overall.

Reply:
commited

Reply:
Thanks